precompile/contract_native_minter.go

```diff
@@ -0,0 +1,129 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package precompile
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	_ StatefulPrecompileConfig = &ContractNativeMinterConfig{}
+	// Singleton StatefulPrecompiledContract for W/R access to the contract deployer allow list.
+	ContractNativeMinterPrecompile StatefulPrecompiledContract = createNativeMinterPrecompile(ContractNativeMinterAddress)
+
+	mintSignature = CalculateFunctionSelector("mint(address,uint256)") // address, amount
+
+	ErrCannotMint = errors.New("non-enabled cannot mint")
```

Comment:
nit: move to new errors file

Reply:
my intention was to encapsulate generic errors in `errors` file and keep precompile-specific errors in their own files.